### PR TITLE
✨ Implement image promotion sync Makefile with core CAPI

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:
Still testing if everything works

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by sbueringer** [comment]
/test pull-cluster-api-provider-vsphere-verify-main

(@randomvariable just fyi, accidentally I created this job without always_run: true, I'll fix that up within the next few days when I audit the ProwJobs)

File: apis/v1alpha3/networkdevicespec_conversion.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-	http://www.apache.org/licenses/LICENSE-2.0
+    http://www.apache.org/licenses/LICENSE-2.0
```
Comment (sbueringer) [inline apis/v1alpha3/networkdevicespec_conversion.go:8]: verify-boilerplate apparently didn't work

File: cloudbuild-nightly.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# See https://cloud.google.com/cloud-build/docs/build-config
+# See https://console.cloud.google.com/gcr/images/k8s-staging-test-infra/global/gcb-docker-gcloud
+timeout: 2700s
```
Comment (sbueringer) [inline cloudbuild-nightly.yaml:3]: cloud build configs for the post-submit jobs in test-infra

File: docs/development.md [hunk 3]
```diff
@@ -52,26 +52,28 @@ you will need the following:
...
-Otherwise, set the `DEV_REGISTRY` environment variable to the OCI repository
+Otherwise, set the `REGISTRY` environment variable to the OCI repository
 you want test images uploaded to.
 
 The following environment variables are supported for configuration:
 
 ```shell
 export VERSION ?= $(shell cat clusterctl-settings.json | jq .config.nextVersion -r)
-export IMAGE_NAME ?= manager
 export PULL_POLICY ?= Always
```
Comment (sbueringer) [inline docs/development.md:62]: Updated the env vars to align them with the ones from core CAPI

File: hack/release.sh [hunk 4]
```diff
@@ -1,224 +0,0 @@
-#!/bin/bash
```
Comment (sbueringer) [inline hack/release.sh:1]: This would fail now anyway. Let's directly switch to image promotion when we merge this PR

File: hack/verify-crds.sh [hunk 5]
```diff
@@ -1,49 +0,0 @@
-#!/bin/bash
```
Comment (sbueringer) [inline hack/verify-crds.sh:1]: This is now part of `make verify-gen`

**Review by srm09 — COMMENTED** [review]
Conditional lgtm barring changes to `Makefile` which I would need to go over again, but looks good on first pass.
/lgtm
/hold

**Review by chrischdi — COMMENTED** [review]
One small nit / optional, because I'm okay to just keep it in sync with CAPI and it should not matter for a release :-)

/lgtm

File: Makefile [hunk 6]
```diff
@@ -229,201 +295,280 @@ e2e-templates: ## Generate e2e cluster templates
...
-
-.PHONY: generate-go
-generate-go: $(CONTROLLER_GEN) $(CONVERSION_GEN) ## Runs Go related generate targets
-	go generate ./...
-	$(CONTROLLER_GEN) \
-		paths=./apis/... \
-		object:headerFile=./hack/boilerplate/boilerplate.generatego.txt
+.PHONY: manager
+manager: ## Build the vsphere manager binary into the ./bin folder
+	go build -trimpath -ldflags "$(LDFLAGS)" -o $(BIN_DIR)/manager sigs.k8s.io/cluster-api-provider-vsphere
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply Makefile]: Thx, fixed

**Comment by sbueringer** [comment]
@randomvariable @srm09 Can we move ahead with this PR?

(sorry for being pushy, just would unblock a bunch of work to get the new release machinery finalized and further ProwJob cleanups)

**Pull request merged**